src/payment/PaymentPage.jsx

```diff
@@ -126,36 +129,47 @@ class PaymentPage extends React.Component {
           aria-label={this.props.intl.formatMessage(messages['payment.section.payment.details.label'])}
           className="col-md-7 pl-md-5"
         >
-          {isFreeBasket ? <PlaceOrderButton /> : (
+          {isFreeBasket ? (
+            <PlaceOrderButton />
+          ) : (
             <React.Fragment>
               <PaymentMethodSelect loading={loading} isBasketProcessing={isBasketProcessing} />
               <PaymentForm />
             </React.Fragment>
-        )}
+          )}
         </section>
       </div>
     );
   }
 
+  renderContent() {
+    const { isEmpty, isRedirect } = this.props;
+
+    if (isEmpty) {
+      return this.renderEmptyMessage();
+    }
+
+    if (isRedirect) {
```

Comment:
Nit: I know you have `!basket.redirect` in `isEmpty`, but it would feel safer(clearer?) to have the isRedirect if statement first.

Reply:
👍Good nit.  Fixing.